lib/kwtool/crawlers.ex

```diff
@@ -0,0 +1,28 @@
+defmodule Kwtool.Crawlers do
+  import Ecto.Query, warn: false
+
+  alias Kwtool.Accounts.Schemas.User
+  alias Kwtool.Crawlers.Schemas.Keyword
+  alias Kwtool.Crawlers.UploadParser
+  alias Kwtool.Repo
+
+  def save_keywords_list(keyword_file, %User{} = user) do
+    keyword_file
+    |> UploadParser.load_from_file()
+    |> Enum.each(fn keyword ->
+      create_keyword(%{
+        phrase: List.first(keyword),
+        status: 0,
+        user_id: user.id
+      })
+    end)
+
+    {:ok, "The keyword file is processed successfully!"}
+  end
+
+  def create_keyword(attrs \\ %{}) do
```

Comment:
maybe `defp` here

Reply:
I am also thinking about this, however, there are 2 things I concern:

- These kinds of methods (CRUD) could be public so we can use them in other places also.
- If we mark it as `private` then we cannot test it.

Should we do change it to private 🤔 ?

Reply:
Yeah, for me, I think we can go with private first as we didn't use it anywhere else now. It is clear when we read it. 

Later we can still move it to public and add test cases if we want to use it from somewhere else. I think the current test for `save_keywords_list` is already covered this function. 

But I'm ok with keeping this if you plan to use it somewhere else 🙆‍♀️  cause you've already added the tests.